pkg/db-repo/repository.go

```diff
@@ -27,6 +27,28 @@ const (
 var operations = []string{Create, Read, Update, Delete, Query}
 var RecordNotFound = gorm.ErrRecordNotFound
 
+type RecordNotFoundError struct {
+	Id      uint
+	ModelId string
+	Err     error
+}
+
+func NewRecordNotFoundError(id uint, modelId string, err error) RecordNotFoundError {
+	return RecordNotFoundError{
+		Id:      id,
+		ModelId: modelId,
+		Err:     err,
+	}
+}
+
+func (e RecordNotFoundError) Error() string {
+	return fmt.Sprintf("could not find model of type %s with id %d", e.ModelId, e.Id)
```

Comment:
you could also add info of the nested error here: `fmt.Sprintf("could not find model of type %s with id %d: %s", e.ModelId, e.Id, e.Err)`